examples/chip-tool/commands/discover/Commands.h

```diff
@@ -22,57 +22,47 @@
 #include "DiscoverCommissionablesCommand.h"
 #include "DiscoverCommissionersCommand.h"
 #include <controller/DeviceAddressUpdateDelegate.h>
-#include <lib/dnssd/Resolver.h>
+#include <lib/address_resolve/AddressResolve.h>
 
-class Resolve : public DiscoverCommand, public chip::Dnssd::OperationalResolveDelegate
+class Resolve : public DiscoverCommand, public chip::AddressResolve::NodeListener
 {
 public:
-    Resolve(CredentialIssuerCommands * credsIssuerConfig) : DiscoverCommand("resolve", credsIssuerConfig) {}
+    Resolve(CredentialIssuerCommands * credsIssuerConfig) : DiscoverCommand("resolve", credsIssuerConfig)
+    {
+        mNodeLookupHandle.SetListener(this);
+    }
 
     /////////// DiscoverCommand Interface /////////
     CHIP_ERROR RunCommand(NodeId remoteId, uint64_t fabricId) override
     {
-        ReturnErrorOnFailure(mDNSResolver.Init(chip::DeviceLayer::UDPEndPointManager()));
-        mDNSResolver.SetOperationalDelegate(this);
-        ChipLogProgress(chipTool, "Dnssd: Searching for NodeId: %" PRIx64 " FabricId: %" PRIx64 " ...", remoteId, fabricId);
-        return mDNSResolver.ResolveNodeId(chip::PeerId().SetNodeId(remoteId).SetCompressedFabricId(fabricId),
-                                          chip::Inet::IPAddressType::kAny);
+        ReturnErrorOnFailure(chip::AddressResolve::Resolver::Instance().Init(&chip::DeviceLayer::SystemLayer()));
+
+        return chip::AddressResolve::Resolver::Instance().LookupNode(
+            chip::AddressResolve::NodeLookupRequest(chip::PeerId().SetNodeId(remoteId).SetCompressedFabricId(fabricId)),
+            mNodeLookupHandle);
     }
 
-    void OnOperationalNodeResolved(const chip::Dnssd::ResolvedNodeData & nodeData) override
+    void OnNodeAddressResolved(const PeerId & peerId, const chip::AddressResolve::ResolveResult & result) override
     {
         char addrBuffer[chip::Transport::PeerAddress::kMaxToStringSize];
 
-        ChipLogProgress(chipTool, "NodeId Resolution: %" PRIu64 " Port: %" PRIu16, nodeData.mPeerId.GetNodeId(), nodeData.mPort);
-        ChipLogProgress(chipTool, "    Hostname: %s", nodeData.mHostName);
-        for (size_t i = 0; i < nodeData.mNumIPs; ++i)
-        {
-            nodeData.mAddress[i].ToString(addrBuffer);
-            ChipLogProgress(chipTool, "    addr %zu: %s", i, addrBuffer);
-        }
-
-        auto retryInterval = nodeData.GetMrpRetryIntervalIdle();
-
-        if (retryInterval.HasValue())
-            ChipLogProgress(chipTool, "   MRP retry interval (idle): %" PRIu32 "ms", retryInterval.Value().count());
-
-        retryInterval = nodeData.GetMrpRetryIntervalActive();
-
-        if (retryInterval.HasValue())
-            ChipLogProgress(chipTool, "   MRP retry interval (active): %" PRIu32 "ms", retryInterval.Value().count());
+        result.address.ToString(addrBuffer);
```

Comment:
But ... this command was in fact meant to be used to log all the addresses being advertised, so it could be used for testing that.  This breaks the command, no?

Reply:
I could see no obvious docks that would say "it lists all addresses". I assumed any mention of "all addresses" is not correct because our mdns implementations cannot seem to guarantee this - at least avahi can only show one it seems (at least when I poked at it I could not get it to give me multiple addresses).

Reply:
This command is meant for testing purposes.  Yes, the docs are not great.  But this was one of the few things we had that we could use to actually see what ips are ending up resolving in practice....  What is the replacement?

And yes, with Avahi the results were wrong.  That should be fixed in the Avahi intergration bits, no?

Reply:
My understanding of the use was `do resolve like chip library does and give me the result` hence my implementation. Is there a need for full results? I was assuming no since we are aware not all libs coud do it. 

I see no very direct replacements so I would be willing to re-work or undo the change (prefer rework really since I would change the name to "resolve-all-addresses").

For debug the resolver logs its own lookups so there is data there beyond just one address and we can enhance that (this command was using LogProgress as well, so maybe no loss in data, but some loss in formatting and we may need to force resolver to log all addresses instead of just new best picks). 

I would like to understand the use case though - who uses this command and where. My change was assuming "this does a resolve of a node and wants the same result as the regular chip library". If instead we want "Log content of DNSSD resolve packets" that works too. In that case I think we should not be using Proxy but rather use the DNSSD interfaces only (as there is no parallelism).

Reply:
tagging @vivien-apple to see how to follow up on this. I am unsure who uses this and where

Reply:
The original goal of this command was to get the ip/port used by the controller to talk to a commissioned device. It was mostly a tool to debug the darwin backend without having to resort to `ChipTool iOS`.

Now I believe it has evolved over time to be used as a debug tool when dealing with mdns advertisement related issues, so a replacement would be useful. About using Proxy versus dnssd interfaces, this is mostly a story of what is the easiest thing to use since the focus for this utility is really on the dnssd data, not on the API.

Reply:
Do I need to take any specific action here to have all addresses reported or is this usable as-is?

I would prefer as little delta as possible - either keep as is or if we need to see all addresses to add logging into the address resolver for each address since that seems to serve the "use for debug" purpose and the final result is useful to "get the ip/port used by the controller".

Reply:
Adding logging into the resolver would be a good idea, I think, and sufficient ofor debugging purposes.

Reply:
created #15653 for this.